### PR TITLE
feat!: update getURL config to use string

### LINKED ISSUE #147: Gatsby v4 compatibility
After upgrading to gatsby 4 and 1.7.0 of the imgix library (which adds gatsby 4 support), I get errors related to invalid config formats, though the format hasn't changed (and still seems to match the readme)

Steps to reproduce the behaviour:

1.  Have the following section in your fields config in gatsby-config.js:

```
fields: [
        {
          nodeType: "Instagram",
          fieldName: "image_imgix",
          getURL: (node) => node.image?.url
        },
        {
          nodeType: "ShopifyProductImages",
          fieldName: "imgix",
          getURL: (node) => node.url 
        },
      ],
```

2.  Build with production node env
3.  Error during "Validating Rendering Engines" of gatsby build:
```
error [@imgix/gatsby] Fatal error during setup: Error: The plugin config is not in the correct format. Errors: Invalid value [{"nodeType":"Instagram","fieldName":"image_imgix"},{"nodeType":"ShopifyProductImages","fieldName":"imgix"}] supplied to : GatsbySourceUrlOptions/fields: (undefined | Array<(({ nodeType: string, fieldName: string } & { getURL: Function }) | ({ nodeType: string, fieldName: string } & { getURLs: Function }))>)/0: undefined,Invalid value undefined supplied to : GatsbySourceUrlOptions/fields: (undefined | Array<(({ nodeType: string, fieldName: string } & { getURL: Function }) | ({ nodeType: string, fieldName: string } & { getURLs: Function }))>)/1: Array<(({ nodeType: string, fieldName: string } & { getURL: Function }) | ({ nodeType: string, fieldName: string } & { getURLs: Function }))>/0: (({ nodeType: string, fieldName: string } & { getURL: Function }) | ({ nodeType: string, fieldName: string } & { getURLs: Function }))/0: ({ nodeType: string, fieldName: string } & { getURL: Function })/1: { getURL: Function }/getURL: Function,Invalid value undefined supplied to : GatsbySourceUrlOptions/fields: (undefined | Array<(({ nodeType: string, fieldName: string } & { getURL: Function }) | ({ nodeType: string, fieldName: string } & { getURLs: Function }))>)/1: Array<(({ nodeType: string, fieldName: string } & { getURL: Function }) | ({ nodeType: string, fieldName: string } & { getURLs: Function }))>/0: (({ nodeType: string, fieldName: string } & { getURL: Function }) | ({ nodeType: string, fieldName: string } & { getURLs: Function }))/1: ({ nodeType: string, fieldName: string } & { getURLs: Function })/1: { getURLs: Function }/getURLs: Function,Invalid value undefined supplied to : GatsbySourceUrlOptions/fields: (undefined | Array<(({ nodeType: string, fieldName: string } & { getURL: Function }) | ({ nodeType: string, fieldName: string } & { getURLs: Function }))>)/1: Array<(({ nodeType: string, fieldName: string } & { getURL: Function }) | ({ nodeType: string, fieldName: string } & { getURLs: Function }))>/1: (({ nodeType: string, fieldName: string } & { getURL: Function }) | ({ nodeType: string, fieldName: string } & { getURLs: Function }))/0: ({ nodeType: string, fieldName: string } & { getURL: Function })/1: { getURL: Function }/getURL: Function,Invalid value undefined supplied to : GatsbySourceUrlOptions/fields: (undefined | Array<(({ nodeType: string, fieldName: string } & { getURL: Function }) | ({ nodeType: string, fieldName: string } & { getURLs: Function }))>)/1: Array<(({ nodeType: string, fieldName: string } & { getURL: Function }) | ({ nodeType: string, fieldName: string } & { getURLs: Function }))>/1: (({ nodeType: string, fieldName: string } & { getURL: Function }) | ({ nodeType: string, fieldName: string } & { getURLs: Function }))/1: ({ nodeType: string, fieldName: string } & { getURLs: Function })/1: { getURLs: Function }/getURLs: Function
```

**Information:**

- imgix/gatsby version: [v1.7.0]

### PR DESCRIPTION
**Pull request by frederickfogerty**
This PR fixes #147. Before this PR, the URL was retrieved in the GraphQL from the underlying data with a function configuration option. This was incompatible with Gatsby v4 (specifically DSG and SSR pages), so the configuration was changed to be a simple string configuration, which was compatible. Furthermore, the name of the config option was changed to better match the new behaviour.

### CODE REVIEW

**Comment by frederickfogerty** [comment]
Current dependencies on/for this PR:
* main
  * **PR #163** <a href="https://app.graphite.dev/github/pr/imgix/gatsby/163" target="_blank">Graphite</a>  (this PR)

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

File: src/modules/gatsby-plugin/gatsby-node.ts [hunk 1]
```diff
@@ -170,27 +170,35 @@ export const createSchemaCustomization: ICreateSchemaCustomizationHook<IImgixGat
                             ? findPossibleURLsInNode(node)
                             : [];
 
+                        const potentialImagesString = (() => {
```
Comment (sherwinski) [inline src/modules/gatsby-plugin/gatsby-node.ts:175]: nice

**Review by sherwinski — APPROVED** [review]
Makes sense to me 👍 
Just wondering if this should be released on beta first in case we want to 1) explore any other major changes before release and 2) get some folks to test it for other edge cases

**Comment by frederickfogerty** [comment]
> 1) explore any other major changes before release

Nothing that I can think of

> 2) get some folks to test it for other edge cases

Sounds like a good idea, I'll merge it into next
